src/features/references/editor/ReferenceDetailsCard.tsx

```diff
@@ -0,0 +1,261 @@
+import { createContext, Dispatch, ReactElement, SetStateAction, useCallback, useContext, useState } from 'react';
+
+import { CloseIcon, EditIcon } from '../../../components/icons';
+import { cx } from '../../../lib/cx';
+import { Author, ReferenceItem } from '../../../types/ReferenceItem';
+
+interface IEditingContext {
+  editing: boolean;
+  setEditing: Dispatch<SetStateAction<boolean>>;
+}
+
+const EditingContext = createContext<IEditingContext>({
+  editing: false,
+  setEditing: () => {
+    console.log('seting edtiting');
+  },
+});
+
+const returnFormatedAuthorsString = (authorsArray: Author[]): string => {
+  let authorsString = '';
+  authorsArray.forEach((a, index) => {
+    authorsString += (index > 0 ? ', ' : '') + a.fullName;
+  });
+  return authorsString;
+};
+
+const getContentString = (content: string | Author[] | undefined, id: string): string => {
+  if (typeof content == 'undefined') {
+    return '';
+  } else if (typeof content != 'string' && id === 'authors') {
+    return returnFormatedAuthorsString(content);
+  } else if (id === 'citationKey') {
+    return '[' + content.toString() + ']';
+  } else {
+    return content.toString();
+  }
+};
+
+const DataTextInput = ({
+  content,
+  id,
+  onChangeHandler,
+}: {
+  content: string;
+  id: string;
+  onChangeHandler: CallableFunction;
+}) => {
+  const [value, setValue] = useState(content);
+
+  return (
+    <input
+      className="w-full border bg-slate-50 px-2 py-0.5"
+      name={id}
+      type="text"
+      value={value}
+      onBlur={(evt) => {
+        onChangeHandler(id, evt.target.value);
+      }}
+      onChange={(evt) => {
+        setValue(evt.target.value);
+      }}
+    />
+  );
+};
+
+const TableDataCell = ({
+  contentData,
+  id,
+  onChangeHandler,
+  editable,
+}: {
+  contentData: string | Author[];
```

Comment:
The type should be `contentData?: string | Author[];` here

Reply:
Now passing an empty string in case of no content.

Reply:
Perfect, that's even better! 👍